firefox-ios/WidgetKit/DownloadManager/DownloadLiveActivity.swift

```diff
@@ -5,9 +5,51 @@
 import WidgetKit
 import ActivityKit
 import SwiftUI
+import Foundation
 
 struct DownloadLiveActivityAttributes: ActivityAttributes {
     struct ContentState: Codable, Hashable {
+        struct Download: Codable, Hashable {
+            var id: UUID
+            var fileName: String
+            var mimeType: String
+            var hasContentEncoding: Bool?
+            var downloadPath: URL
+
+            var totalBytesExpected: Int64?
+            var bytesDownloaded: UInt64
+            var isComplete: Bool
+        }
+        var downloads: [Download]
+
+        var completedDownloads: Int {
+            return downloads.filter { $0.isComplete }.count
+        }
+
+        var totalDownloads: Int {
+            return downloads.count
+        }
+
+        func getTotalProgress() -> Double {
```

Comment:
Might be helpful to make totalBytesExpected and totalBytesDownloaded computed variables. Since the live activity will be using basically what totalBytesExpected and totalBytesDownloaded are to show the progress e.g.("10MB of 20MB") it would probably be helpful to include the two as computed properties. This way the UI element can more easily access these values.